mistakedetection.tests/src/ca/mcgill/sel/mistakedetection/tests/MistakeDetectionPerformanceAnalysis.java

```diff
@@ -264,6 +342,298 @@ public void testStudentSolution20() {
+        new Object[] {"Extra Elements", "=SUM(IF(D" + extraElemsStart + ":D" + id + ">E" + extraElemsStart + ":E" + id
+            + ",D" + extraElemsStart + ":D" + id + ",E" + extraElemsStart + ":E" + id + "))"});
+    studentData.put(String.valueOf(nId++), new Object[] {"Additional false negatives", 0});
+    studentData.put(String.valueOf(nId++), new Object[] {"Additional false positive", 0});
+    studentData.put(String.valueOf(nId++), new Object[] {"Total elements",
+        "=B" + totalInstElemsIndex + "+B" + studExtraIndex + "+B" + addFalseNegIndex + "+B" + addFalsePosIndex});
+    studentData.put(String.valueOf(nId++), new Object[] {"Total number of verdicts",
+        "=SUM(IF(D2:D" + instructorElems + "=2, 1, IF(E2:E" + instructorElems + "=2, 1,0))) + B" + totalElemsIndex});
+    studentData.put(String.valueOf(nId++), new Object[] {"TN+TP+FP+FN", "=SUM(C" + tnIndex + ":C" + fnIndex + ")"});
+
+    studentData.put(String.valueOf(nId++), new Object[] {""});
+    studentData.put(String.valueOf(nId++), new Object[] {"", "", "Actual Vs MDS"});
+
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"Correct Identification % (mistakes)", "", "=C" + tpIndex + "/E" + totalMistakeindex + "*100"});
+    studentData.put(String.valueOf(nId++), new Object[] {"Correct Identification % (verdicts)", "",
+        "=((B" + totalNumbVerdictsIndex + "-F" + MDSvsActualIndex + ")/B" + totalNumbVerdictsIndex + ")*100"});
+    studentData.put(String.valueOf(nId++), new Object[] {"True Negative", "", "=SUM(IF(D2:D" + instructorElems
+        + "=0, IF(E2:E" + instructorElems + "=0, 1,0), 0 )) + (102 - B" + instElemsIndex + ")"});
+    studentData.put(String.valueOf(nId++), new Object[] {"True Positive", "",
+        "=SUM(IF(D2:D" + id + ">0, IF(D2:D" + id + "> E2:E" + id + ", E2:E" + id + ", D2:D" + id + "), 0 ))"});
+    studentData.put(String.valueOf(nId++), new Object[] {"False Positive", "",
+        "=SUM(IF(E2:E" + id + "> D2:D" + id + ", E2:E" + id + " -D2:D" + id + "), 0 )+B" + addFalsePosIndex});
+    studentData.put(String.valueOf(nId++), new Object[] {"False Negative", "",
+        "=SUM(IF(D2:D" + id + ">E2:E" + id + ", D2:D" + id + " - E2:E" + id + "), 0 )+B" + addFalseNegIndex});
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"Recall (TP / (TP + FN))", "", "=(C" + tpIndex + "/(C" + tpIndex + "+C" + fnIndex + "))"});
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"Precision (TP / (TP + FP))", "", "=(C" + tpIndex + "/(C" + tpIndex + "+C" + fpIndex + "))"});
+    
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"F1 (2 * Precision * Recall / (Precision + Recall))", "", "=(2*C"+precisionIndex+"*C"+recallIndex+")/(C"+precisionIndex+"+C"+recallIndex+")"});
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"F2 (5 * Precision * Recall) / (4 * Precision + Recall)", "", "=(5*C"+precisionIndex+"*C"+recallIndex+")/(4*C"+precisionIndex+"+C"+recallIndex+")"});
```

Comment:
A lot of the lines above are repetitive and prone to typos, so extract common logic to helper functions

Reply:
yes, I agree, because of time constraints I did it this way.